### PR TITLE
Heal 99

### PR DESCRIPTION
**Pull request by Shruthi23**
HEAL 99 - Add mapping between doctor and healthcare provider - Backend Smart contracts

HEAL 100 -Smart contracts to reject and accept the request raised by a doctor to join the network

Hardcoded the healthcare address for now. Replace the health care provider addresses in UpdateProfile.tsx with the healthcare address on your blockchain.

### CODE REVIEW

File: contracts/HealthBlock.sol [hunk 1]
```diff
@@ -172,10 +178,59 @@ contract HealthBlock {
...
+        uint8 idx =0;
+        bool found = false;
+        for(uint8 i =0;i< providerToDoctorRequests[_providerAddress].length;i++){
+            if(providerToDoctorRequests[_providerAddress][i].doctorAddr == _doctorAddress){
+                idx = i;
+                break;
+                found = true;
+ }
+        }
+
```
Comment (varunalla) [inline contracts/HealthBlock.sol:190]: does this scale? just make sure this does not take time. a mapping is better positioned for such use cases.

File: client/src/providers/HealthProvider.tsx [hunk 2]
```diff
@@ -284,21 +293,69 @@ export const HealthProvider: React.FC<Props> = ({ children, ...props }) => {
...
+  ) => {
+    try {
+      const web3modal = new Web3Modal();
+      const connection = await web3modal.connect();
+      const provider = new ethers.providers.Web3Provider(connection);
+      const signer = provider.getSigner();
+      const contract: HealthBlock = fetchContract(signer);
+      const update = await contract.raiseDoctorToProviderRequest(hcAddress, docAddress, doctorName);
+    } catch (err: any) {
+      setError(`Error Loading Health Contract ${err}`);
```
Comment (varunalla) [inline client/src/providers/HealthProvider.tsx:355]: please dont use this format. these are hooks, return the error to the calling component and let them handle the error.

**Review by varunalla — APPROVED** [review]
lgtm! except the error handling. please throw the error to the calling component.

File: contracts/HealthBlock.sol (hunk 1 above)
Comment (Shruthi23) [reply contracts/HealthBlock.sol:190]: Thank you for the review. Addressing it in the next pr.

File: client/src/providers/HealthProvider.tsx (hunk 2 above)
Comment (Shruthi23) [reply client/src/providers/HealthProvider.tsx:355]: Thank you for the review. I will update it in the next PR.

**Pull request merged**